app/weddingapp/models.py

```diff
@@ -1,3 +1,60 @@
-# from django.db import models
+"""
+Database models
+"""
+from django.db import models
+from django.contrib.auth.models import (
+    AbstractBaseUser,
+    BaseUserManager,
+    PermissionsMixin
+)
 
-# Create your models here.
+
+class UserManager(BaseUserManager):
+    """Manager for users"""
+
+    def create_user(self, username, password=None, email=None, **extra_fields):
+        """
+        Create, save and return a new user with given username and password
+        """
+        if not username:
+            raise ValueError("Please provide an username.")
+        user = self.model(username=username,
+                          email=self.normalize_email(email), **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+
+        return user
+
+    def create_superuser(self, username, password):
+        """Create and return a new superuser"""
+        user = self.create_user(username, password)
+        user.is_staff = True
+        user.is_superuser = True
+        user.save(using=self._db)
+
+        return user
+
+    def get_by_natural_key(self, username):
+        """Make username case insensitive"""
+        case_insensitive_username_field = '{}__iexact'.format(
+            self.model.USERNAME_FIELD)
+
+        return self.get(**{case_insensitive_username_field: username})
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    """User in the system"""
+    username = models.CharField(
+        "first name + last name",
```

Comment:
I don't know exactly where this string is gonna be used, but when you login in admin page it appears in the place of "Username", I think the correct is to show "Username" itself.

Reply:
If it is a placeholder, maybe empty then. Users tend to ignore spaces when there is something written already. Later, we can put a tooltip.

Reply:
It is not a placeholder. It is the name of the field that will appear in the login page, like Bruno said. I changed to "Username".